src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -692,6 +692,14 @@ trait OptimisticTransactionImpl extends TransactionalWrite with SQLMetricsReport
     attemptVersion
   }
 
+  /**
+   * Used to filter collections of `FileAction`s with `dataChange=false`
```

Comment:
nit: there's no collections of `FileAction`s so it's misleading.

Reply:
Yes, I realize how how this phrasing doesn’t say exactly what I wanted to describe. My intention was to point out how this function is *used*, and not what it *does*. How about this alternative description:

This function is used to determine if the given `FileAction` should be considered when checking for conflicts based on the value of `dataChange` and the configuration toggle `DeltaSQLConf.CONCURRENT_COMPACTION_CONFLICT_DETECTION`